agile-app/src/app/map/page.tsx

```diff
@@ -55,37 +71,53 @@ export default function Map() {
                 locationAmountDict[event.location.name] = 1
             }
         }
+        maxCrimeValue = Math.max(...Object.values(locationAmountDict));
         return locationAmountDict;
+
     }
 
+    
+
+    
     /**
      * Determines the color style for rendering GeoJSON features based on the density of crime events.
      * 
      * @param density - The density of crime events.
      * @returns The color code for rendering the feature.
      */
 
+    // function setMaxCrimeValue() {
+    //     maxCrimeValue = Math.max(...Object.values(getEventsOnType(selectedOptionCrime)));
+      
+    // }
+
     function getColor(density : number) {
```

Comment:
Make this more extendable by using a loop that iterates over intervals or a map to define the colors. You could even move out the color intervals and declare it as a constant outside the function.
```suggestion
    function getColor(density : number) {
        let maxCrimeSixth = maxCrimeValue/6;
        return (density > maxCrimeValue-1 ? '#b30000' :
            density > maxCrimeSixth*5 ? '#e34a33' :
                density > maxCrimeSixth*4 ? '#fc8d59' :
                    density > maxCrimeSixth*3 ? '#fdbb84' :
                        density > maxCrimeSixth*2 ? '#fdd49e' :
                            density > maxCrimeSixth ? '#fef0d9' :
                                '#FFFFFF')
    }
``` -> ```suggestion
    function getColor(density : number) {
        const colorIntervals: [number, string][] = [
        [6, '#b30000'],
        [5, '#e34a33'],
        [4, '#fc8d59'],
        [3, '#fdbb84'],
        [2, '#fdd49e'],
        [1, '#fef0d9'],
        [0, '#FFFFFF']
        ];
        for (const [threshold, color] of colorMap) {
            if (density > maxCrimeSixth * threshold) {
                return color;
            }
        }
    }
```

Reply:
This has now been fixed